### PR TITLE
bug fix: create/update CRD panic when mutating webhook configured

### LINKED ISSUE #61355: Unable to create/update CRD when mutating webhook configured
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:

I spun up a new K8s 1.9 on-prem cluster and added a mutating webhook:
```yaml
apiVersion: admissionregistration.k8s.io/v1beta1
kind: MutatingWebhookConfiguration
metadata:
  name: webhook-name
webhooks:
  - name: webhook-name
    failurePolicy: Fail
    clientConfig:
      url: https://127.0.0.1:12345/auth/admit
      caBundle: ...
    rules:
    - operations:
      - "CREATE"
      - "UPDATE"
      apiGroups:
      - "*"
      apiVersions:
      - "*"
      resources:
      - "*"
```

I then tried to create a CRD:
```
apiVersion: apiextensions.k8s.io/v1beta1
kind: CustomResourceDefinition
metadata:
  name: crontabs.stable.example.com
spec:
  group: stable.example.com
  version: v1
  scope: Namespaced
  names:
    plural: crontabs
    singular: crontab
    kind: CronTab
    shortNames:
    - ct
```

and got the error:
```
Error from server (InternalError): error when creating "crd.yml": Internal error occurred: no kind "CustomResourceDefinition" is registered for version "apiextensions.k8s.io/v1beta1"
```

If the CRD already exists, I get this error instead:
```
Error from server (InternalError): error when applying patch:
{"metadata":{"annotations":{},"namespace":""},"spec":{"names":{}}}
to:
&{0xc420105080 0xc420178f50  crontabs.stable.example.com crd.yml 0xc4207597b8 628296 false}
for: "crd.yml": Internal error occurred: no kind "CustomResourceDefinition" is registered for version "apiextensions.k8s.io/v1beta1"
```

**What you expected to happen**:

```
customresourcedefinition "crontabs.stable.example.com" created
```
or
```
customresourcedefinition "crontabs.stable.example.com" configured
```

**How to reproduce it (as minimally and precisely as possible)**:
 - Create a mutating webhook (like above. the endpoint doesn't need to exist as it is never hit).
 - Try to create a CRD (like above)
 - Failure

**Anything else we need to know?**:
Logs have this:
<details><summary>Click to expand</summary><p>

```
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: goroutine 6378189 [running]:
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/httplog.(*respLogger).recordStatus(0xc42f274000, 0x1f4)
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/httplog/httplog.go:207 +0xdd
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/httplog.(*respLogger).WriteHeader(0xc42f274000, 0x1f4)
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/httplog/httplog.go:186 +0x35
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters.(*baseTimeoutWriter).WriteHeader(0xc428474640, 0x1f4)
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters/timeout.go:188 +0xac
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/metrics.(*ResponseWriterDelegator).WriteHeader(0xc42a345170, 0x1f4)
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go:258 +0x45
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters.SerializeObject(0x3a5b84d, 0x10, 0x7fdb8c78bea0, 0xc4236a1980, 0x8da19e0, 0xc420c9cd18, 0xc42925f200, 0x1f4, 0x8d89b60, 0xc42da4b5f0)
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters/writers.go:95 +0x8d
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters.WriteObjectNegotiated(0x7fdb8c5d49b8, 0xc42a3454d0, 0x8da4020, 0xc420f1c600, 0x3a68bbb, 0x14, 0x3a487e2, 0x7, 0x8da19e0, 0xc420c9cd18, ...)
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters/writers.go:123 +0x32b
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters.ErrorNegotiated(0x7fdb8c5d49b8, 0xc42a3454d0, 0x8d77e60, 0xc42da4b560, 0x8da4020, 0xc420f1c600, 0x3a68bbb, 0x14, 0x3a487e2, 0x7, ...)
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters/writers.go:142 +0x163
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/handlers.(*RequestScope).err(0xc4236f0140, 0x8d77e60, 0xc42da4b560, 0x8da19e0, 0xc420c9cd18, 0xc42925f200)
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/handlers/rest.go:67 +0x10c
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/handlers.createHandler.func1(0x8da19e0, 0xc420c9cd18, 0xc42925f200)
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/handlers/create.go:101 +0x1645
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints.restfulCreateResource.func1(0xc42a345080, 0xc434d2a300)
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/installer.go:1080 +0xd5
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/metrics.InstrumentRouteFunc.func1(0xc42a345080, 0xc434d2a300)
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go:199 +0x208
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: k8s.io/kubernetes/vendor/github.com/emicklei/go-restful.(*Container).dispatch(0xc420edd3b0, 0x7fdb8c5899f8, 0xc420f07bc8, 0xc42925f200)
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/emicklei/go-restful/container.go:277 +0xb18
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: k8s.io/kubernetes/vendor/github.com/emicklei/go-restful.(*Container).Dispatch(0xc420edd3b0, 0x7fdb8c5899f8, 0xc420f07bc8, 0xc42925f200)
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/emicklei/go-restful/container.go:199 +0x57
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server.director.ServeHTTP(0x3a74477, 0x17, 0xc420edd3b0, 0xc42036ed90, 0x7fdb8c5899f8, 0xc420f07bc8, 0xc42925f200)
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/handler.go:153 +0x4e0
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server.(*director).ServeHTTP(0xc420a12360, 0x7fdb8c5899f8, 0xc420f07bc8, 0xc42925f200)
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: <autogenerated>:1 +0x75
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux.(*pathHandler).ServeHTTP(0xc426684300, 0x7fdb8c5899f8, 0xc420f07bc8, 0xc42925f200)
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux/pathrecorder.go:254 +0x407
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux.(*PathRecorderMux).ServeHTTP(0xc42011b490, 0x7fdb8c5899f8, 0xc420f07bc8, 0xc42925f200)
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux/pathrecorder.go:234 +0xa1
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server.director.ServeHTTP(0x3a56112, 0xe, 0xc42022f0e0, 0xc42011b490, 0x7fdb8c5899f8, 0xc420f07bc8, 0xc42925f200)
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/handler.go:161 +0x6ad
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server.(*director).ServeHTTP(0xc4208f1b80, 0x7fdb8c5899f8, 0xc420f07bc8, 0xc42925f200)
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: <autogenerated>:1 +0x75
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: k8s.io/kubernetes/vendor/k8s.io/kube-aggregator/pkg/apiserver.(*proxyHandler).ServeHTTP(0xc42ecb9600, 0x7fdb8c5899f8, 0xc420f07bc8, 0xc42925f200)
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/kube-aggregator/pkg/apiserver/handler_proxy.go:91 +0x15f
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux.(*pathHandler).ServeHTTP(0xc42eeaf240, 0x7fdb8c5899f8, 0xc420f07bc8, 0xc42925f200)
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux/pathrecorder.go:248 +0x26d
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux.(*PathRecorderMux).ServeHTTP(0xc423304460, 0x7fdb8c5899f8, 0xc420f07bc8, 0xc42925f200)
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux/pathrecorder.go:234 +0xa1
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server.director.ServeHTTP(0x3a58df0, 0xf, 0xc420613290, 0xc423304460, 0x7fdb8c5899f8, 0xc420f07bc8, 0xc42925f200)
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/handler.go:161 +0x6ad
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server.(*director).ServeHTTP(0xc429832240, 0x7fdb8c5899f8, 0xc420f07bc8, 0xc42925f200)
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: <autogenerated>:1 +0x75
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/filters.WithAuthorization.func1(0x7fdb8c5899f8, 0xc420f07bc8, 0xc42925f200)
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/filters/authorization.go:51 +0x37d
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: net/http.HandlerFunc.ServeHTTP(0xc422bba8c0, 0x7fdb8c5899f8, 0xc420f07bc8, 0xc42925f200)
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: /usr/local/go/src/net/http/server.go:1918 +0x44
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters.WithMaxInFlightLimit.func1(0x7fdb8c5899f8, 0xc420f07bc8, 0xc42925f200)
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters/maxinflight.go:165 +0x42a
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: net/http.HandlerFunc.ServeHTTP(0xc427375180, 0x7fdb8c5899f8, 0xc420f07bc8, 0xc42925f200)
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: /usr/local/go/src/net/http/server.go:1918 +0x44
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/filters.WithImpersonation.func1(0x7fdb8c5899f8, 0xc420f07bc8, 0xc42925f200)
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/filters/impersonation.go:49 +0x203a
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: net/http.HandlerFunc.ServeHTTP(0xc422bba960, 0x7fdb8c5899f8, 0xc420f07bc8, 0xc42925f200)
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: /usr/local/go/src/net/http/server.go:1918 +0x44
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/filters.WithAuthentication.func1(0x7fdb8c5899f8, 0xc420f07bc8, 0xc42925f200)
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/filters/authentication.go:79 +0x2b1
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: net/http.HandlerFunc.ServeHTTP(0xc422bbaa50, 0x7fdb8c5899f8, 0xc420f07bc8, 0xc42925f200)
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: /usr/local/go/src/net/http/server.go:1918 +0x44
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/request.WithRequestContext.func1(0x7fdb8c5899f8, 0xc420f07bc8, 0xc42925f200)
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/request/requestcontext.go:110 +0xcb
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: net/http.HandlerFunc.ServeHTTP(0xc429832280, 0x7fdb8c5899f8, 0xc420f07bc8, 0xc42925f200)
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: /usr/local/go/src/net/http/server.go:1918 +0x44
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters.(*timeoutHandler).ServeHTTP.func1(0xc429832300, 0x8da5aa0, 0xc420f07bc8, 0xc42925f200, 0xc42c529380)
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters/timeout.go:93 +0x8d
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: created by k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters.(*timeoutHandler).ServeHTTP
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters/timeout.go:92 +0x1ab
Mar 15 20:50:32 ip-10-0-121-56 kube-apiserver[1499]: logging error output: "{\"kind\":\"Status\",\"apiVersion\":\"v1\",\"metadata\":{},\"status\":\"Failure\",\"message\":\"Internal error occurred: no kind \\\"CustomResourceDefinition\\\" is registered for version \\\"apiextensions.k8s.io/v1beta1\\\"\",\"reason\":\"InternalError\",\"details\":{\"causes\":[{\"message\":\"no kind \\\"CustomResourceDefinition\\\" is registered for version \\\"apiextensions.k8s.io/v1beta1\\\"\"}]},\"code\":500}\n"
```
</p>
</details>
<br />

**Environment**:
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.3", GitCommit:"d2835416544f298c919e2ead3be3d0864b52323b", GitTreeState:"clean", BuildDate:"2018-02-07T12:22:21Z", GoVersion:"go1.9.2", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.3", GitCommit:"d2835416544f298c919e2ead3be3d0864b52323b", GitTreeState:"clean", BuildDate:"2018-02-07T11:55:20Z", GoVersion:"go1.9.2", Compiler:"gc", Platform:"linux/amd64"}
```

tags (because finding info about CRDs is hard on github):
custom resource definition
customresourcedefinition
customresourcedefinitions

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
apiextentions apiserver use its own scheme to create a admission chain before creating the apiextensions apiserver config.
Fixes #61355

**Special notes for your reviewer**:

**Release note**:

```release-note
Fix panic create/update CRD when mutating/validating webhook configured.
```

### CODE REVIEW

File: cmd/kube-apiserver/app/apiextensions.go [hunk 1]
```diff
@@ -23,17 +23,26 @@ import (
...
 	"k8s.io/kubernetes/cmd/kube-apiserver/app/options"
 )
 
-func createAPIExtensionsConfig(kubeAPIServerConfig genericapiserver.Config, externalInformers kubeexternalinformers.SharedInformerFactory, commandOptions *options.ServerRunOptions) (*apiextensionsapiserver.Config, error) {
+func createAPIExtensionsConfig(kubeAPIServerConfig genericapiserver.Config, externalInformers kubeexternalinformers.SharedInformerFactory, pluginInitializers []admission.PluginInitializer, commandOptions *options.ServerRunOptions) (*apiextensionsapiserver.Config, error) {
 	// make a shallow copy to let us twiddle a few things
 	// most of the config actually remains the same.  We only need to mess with a couple items related to the particulars of the apiextensions
 	genericConfig := kubeAPIServerConfig
 
+	// override genericConfig.AdmissionControl with apiextensions' scheme
```
Comment (sttts) [inline cmd/kube-apiserver/app/apiextensions.go]: can you write why we need this?

Btw, I would bet that we need the same in the aggregator. APIService is not in the legacyscheme (and shouldn't be).

Comment (hzxuzhonghu) [reply cmd/kube-apiserver/app/apiextensions.go]: Sure， I missed it.

File: cmd/kube-apiserver/app/server.go [hunk 2]
```diff
@@ -269,25 +269,25 @@ func CreateNodeDialer(s *options.ServerRunOptions) (tunneler.Tunneler, *http.Tra
 }
 
 // CreateKubeAPIServerConfig creates all the resources for running the API server, but runs none of them
-func CreateKubeAPIServerConfig(s *options.ServerRunOptions, nodeTunneler tunneler.Tunneler, proxyTransport *http.Transport) (*master.Config, informers.SharedInformerFactory, clientgoinformers.SharedInformerFactory, *kubeserver.InsecureServingInfo, aggregatorapiserver.ServiceResolver, error) {
+func CreateKubeAPIServerConfig(s *options.ServerRunOptions, nodeTunneler tunneler.Tunneler, proxyTransport *http.Transport) (*master.Config, informers.SharedInformerFactory, clientgoinformers.SharedInformerFactory, *kubeserver.InsecureServingInfo, aggregatorapiserver.ServiceResolver, []admission.PluginInitializer, error) {
```
Comment (caesarxuchao) [inline cmd/kube-apiserver/app/server.go]: Break the line?

File: cmd/kube-apiserver/app/aggregator.go [hunk 3]
```diff
@@ -49,11 +50,27 @@ import (
...
+	serviceResolver aggregatorapiserver.ServiceResolver,
+	proxyTransport *http.Transport,
+	pluginInitializers []admission.PluginInitializer,
+) (*aggregatorapiserver.Config, error) {
 	// make a shallow copy to let us twiddle a few things
 	// most of the config actually remains the same.  We only need to mess with a couple items related to the particulars of the aggregator
 	genericConfig := kubeAPIServerConfig
 
+	// override genericConfig.AdmissionControl with kube-aggregator's scheme,
+	// because aggregator apiserver should use its own scheme to convert its own resource.
```
Comment (caesarxuchao) [inline cmd/kube-apiserver/app/aggregator.go]: Does aggregator itself have an admission chain? I thought aggregator delegates the admission to each apiservers.

**Comment by caesarxuchao** [comment]
Could you also add an e2e test to webhook.go to verify that it can handle CRD now?

File: cmd/kube-apiserver/app/aggregator.go (hunk 3 above)
Comment (hzxuzhonghu) [reply cmd/kube-apiserver/app/aggregator.go]: I think yes,  APIService handled by aggregator.

**Comment by hzxuzhonghu** [comment]
>Could you also add an e2e test to webhook.go to verify that it can handle CRD now?

Yes.

File: cmd/kube-apiserver/app/server.go (hunk 2 above)
Comment (hzxuzhonghu) [reply cmd/kube-apiserver/app/server.go]: return a new param admission pluginInitializers

File: cmd/kube-apiserver/app/aggregator.go (hunk 3 above)
Comment (sttts) [reply cmd/kube-apiserver/app/aggregator.go]: yes, APIService is served by the aggregator. It's a normal apiserver.

File: cmd/kube-apiserver/app/server.go (hunk 2 above)
Comment (caesarxuchao) [reply cmd/kube-apiserver/app/server.go]: I meant that if you could break this long line into multiple lines, like what you did [here](https://github.com/kubernetes/kubernetes/pull/61404/files/85e69c240c449762d50b0e2fa9687f7efd1bde84#diff-13674dbb1dbd04c901cd058d402e71a3R33).

**Comment by caesarxuchao** [comment]
lgtm except for a cosmetic nit.

File: cmd/kube-apiserver/app/server.go (hunk 2 above)
Comment (hzxuzhonghu) [reply cmd/kube-apiserver/app/server.go]: ic, forgive my poor English

**Comment by SerialVelocity** [comment]
@hzxuzhonghu Thank you so much for such a quick fix for this issue! Do you know whether this will be backported to 1.9.x?

**Comment by hzxuzhonghu** [comment]
@SerialVelocity we can cherry-pick onto 1.9

**Comment by hzxuzhonghu** [comment]
@sttts for another look

**Comment by hzxuzhonghu** [comment]
@deads2k ptal

**Comment by hzxuzhonghu** [comment]
ping @deads2k @caesarxuchao @sttts for approval

**Comment by caesarxuchao** [comment]
Can you squash the 1st, 2nd, and 4th commits into one?

**Comment by caesarxuchao** [comment]
I'll leave it to sttts to approve. I can't approve test/

**Comment by hzxuzhonghu** [comment]
squashed

**Comment by hzxuzhonghu** [comment]
rebased

File: cmd/kube-apiserver/app/server.go [hunk 4]
```diff
@@ -445,7 +473,7 @@ func BuildGenericConfig(s *options.ServerRunOptions, proxyTransport *http.Transp
 	if err != nil {
 		kubeAPIVersions := os.Getenv("KUBE_API_VERSIONS")
 		if len(kubeAPIVersions) == 0 {
-			return nil, nil, nil, nil, nil, fmt.Errorf("failed to create clientset: %v", err)
+			return nil, nil, nil, nil, nil, nil, fmt.Errorf("failed to create clientset: %v", err)
```
Comment (sttts) [inline cmd/kube-apiserver/app/server.go]: no lastErr logic here?

**Comment by sttts** [comment]
One nit. Overall lgtm.

/approve

File: cmd/kube-apiserver/app/server.go (hunk 4 above)
Comment (hzxuzhonghu) [reply cmd/kube-apiserver/app/server.go]: deal with it right now

**Comment by hzxuzhonghu** [comment]
@sttts Done

File: test/e2e/apimachinery/webhook.go [hunk 5]
```diff
@@ -1121,3 +1129,92 @@ func testMutatingCRDWebhook(f *framework.Framework, crd *apiextensionsv1beta1.Cu
...
+				Plural:   testcrd.GetPluralName(),
+				Singular: testcrd.Name,
+				Kind:     testcrd.Kind,
+				ListKind: testcrd.GetListName(),
+			},
+			Scope: apiextensionsv1beta1.NamespaceScoped,
+		},
+	}
+
+	//create CRD
```
Comment (sttts) [inline test/e2e/apimachinery/webhook.go]: nit: `// create`

File: cmd/kube-apiserver/app/apiextensions.go [hunk 6]
```diff
@@ -23,17 +23,32 @@ import (
...
+	externalInformers kubeexternalinformers.SharedInformerFactory,
+	pluginInitializers []admission.PluginInitializer,
+	commandOptions *options.ServerRunOptions,
+) (*apiextensionsapiserver.Config, error) {
 	// make a shallow copy to let us twiddle a few things
 	// most of the config actually remains the same.  We only need to mess with a couple items related to the particulars of the apiextensions
 	genericConfig := kubeAPIServerConfig
 
+	// override genericConfig.AdmissionControl with apiextensions' scheme,
+	// because apiextentions apiserver should use its own scheme to convert resource.
```
Comment (sttts) [inline cmd/kube-apiserver/app/apiextensions.go]: nit: resources

File: cmd/kube-apiserver/app/aggregator.go (hunk 3 above)
Comment (sttts) [inline cmd/kube-apiserver/app/aggregator.go]: nit: resources

Comment (hzxuzhonghu) [reply cmd/kube-apiserver/app/aggregator.go]: done

File: test/e2e/apimachinery/webhook.go (hunk 5 above)
Comment (hzxuzhonghu) [reply test/e2e/apimachinery/webhook.go]: done

File: cmd/kube-apiserver/app/apiextensions.go (hunk 6 above)
Comment (hzxuzhonghu) [reply cmd/kube-apiserver/app/apiextensions.go]: done

**Pull request merged**